lib/js/jslib.js

```diff
@@ -3006,97 +3142,76 @@ function get_process_id(pid) {
   } else if (_is_valid_server_pid(pid)) {
     return pid["_serverPid"];
   } else {
-    DEBUG.assert(false, "Invalid PID in get_process_id");
+    _$Debug.assert(false, "Invalid PID in get_process_id");
     return;
   }
 }
 
 function _Send(pid, msg) {
   //  _dump(_mailboxes)
   /* First, check to see whether we are sending to a well-formed PID */
-  var valid_server_pid = _is_valid_server_pid(pid);
-  var valid_client_pid = _is_valid_client_pid(pid);
-  DEBUG.assert(
+  let valid_server_pid = _is_valid_server_pid(pid);
+  let valid_client_pid = _is_valid_client_pid(pid);
+  _$Debug.assert(
     valid_server_pid || valid_client_pid,
     "Malformed PID in _Send: neither client nor server PID"
   );
 
   if (valid_server_pid) {
-    WEBSOCKET.sendRemoteServerMessage(pid._serverPid, msg);
+    _$Websocket.sendRemoteServerMessage(pid._serverPid, msg);
   } else if (valid_client_pid) {
     if (get_client_id(pid) == _client_id) {
       // Local send
-      DEBUG.debug("sending message ", msg, " to pid ", pid._clientPid);
-      var client_pid = pid._clientPid;
-      LINKS.deliverMessage(client_pid, msg);
+      _$Debug.debug("sending message ", msg, " to pid ", pid._clientPid);
+      let client_pid = pid._clientPid;
+      _$Links.deliverMessage(client_pid, msg);
     } else {
       // Remote send
-      WEBSOCKET.sendRemoteClientMessage(get_client_id(pid), pid._clientPid, msg);
+      _$Websocket.sendRemoteClientMessage(get_client_id(pid), pid._clientPid, msg);
     }
   }
   //_dumpSchedStatus();
   return;
 }
 
 function Send(pid, msg, kappa) {
-  return _applyCont(kappa, _Send(pid, msg));
-}
-
-function _SendWrapper(env) {  // necessary wrapper for server->client calls
-  return Send;
+  return _$K.apply(kappa, _Send(pid, msg));
 }
 
 function _dictlength(x) {
-  var length = 0;
+  let length = 0;
   for (let prop in x) {
     length++;
   }
   return length;
 }
 
-function _block_proc(pid, its_cont) {
-  DEBUG.debug("Blocking ", pid);
-  _blocked_procs[pid] = its_cont;
-  // discard stack
-  return;
-}
-
 // recv
 //   recv is an unusual library function that may capture the
 //   continuation; hence there is no _recv form (direct-style).
 function recv(kappa) {
-  DEBUG.assert(
+  _$Debug.assert(
     arguments.length == 1,
     'recv received ' + arguments.length + ' arguments, expected 1'
   );
-  //DEBUG.assert(_current_pid != _mainPid,
-  //           "Cannot call recv() in main process.");
-  DEBUG.assert(
```

Comment:
why has this assertion been deleted?

Reply:
I remember now. It's because `mainPid` is a hidden property now. I could expose it again to keep this assertion alive.